src/db/models.rs

```diff
@@ -8,11 +8,18 @@ use std::borrow::Cow;
 use chrono::DateTime;
 use chrono::offset::Utc;
 use diesel::prelude::*;
+use pwhash::bcrypt;
 use db::schema::{accounts, follows, statuses, users};
 use db::Connection;
-use pwhash::bcrypt;
+use diesel::result::Error as DieselError;
 use {BASE_URL, DOMAIN};
 
+/// Type representing the result of a database operation.
+///
+/// The nested `Option<T>` in the `Result` allows a semantic distinction
+/// bewteen "query error" and "no records returned".
+type DatabaseResult<T> = Result<Option<T>, DieselError>;
```

Comment:
I wonder if this is what we want if we're returning multiple rows? Option<Vec<_>> doesn't seem quite right to me, because a Vec already knows what nothingness is.

Reply:
Hmmmmmmmmm, that's a _very_ good point. Maybe we should turn `DatabaseResult` into
```rust
type DBResult<T> = Result<T, DieselError>;
```
and explicitly note that it wraps an `Option<T>` in the signatures of individual functions?

Reply:
Yeah, I like that.  On the Diesel issue tracker, there seems to be interest in bringing back iterators for lazy-loading result sets, and `Result<T, DieselError>` accounts for that case as well, I think.

Reply:
Wait, isn't this just http://docs.diesel.rs/diesel/result/type.QueryResult.html ?

Reply:
@ilianaw Yep. We could definitely just use that.